eox_nelp/migrations/0001_initial.py

```diff
@@ -95,6 +95,26 @@ class Migration(migrations.Migration):
                 ('org', models.CharField(blank=True, max_length=500, null=True))
             ],
         ),
+        migrations.CreateModel(
```

Comment:
what you are creating a a backend if you are creating the model ?

Reply:
Because the backend allows me to load the class that passes the data to the test db.
https://github.com/eduNEXT/eox-nelp/blob/f398520f00a55ef12283d7563ee2624b49dbf2b6/eox_nelp/stats/tests/tests_metrics.py#L273
https://github.com/eduNEXT/eox-nelp/blob/f398520f00a55ef12283d7563ee2624b49dbf2b6/eox_nelp/edxapp_wrapper/test_backends/certificates_m_v1.py#L32-L58

Reply:
generally we don't duplicate models from edx-platform, why this case is so especial ?

Reply:
Actually is because in the test I use the model to generate the certificates in `db`. So the mainly reason is that, as you can see there course_overview also was recreated, but with less fields
https://github.com/eduNEXT/eox-nelp/blob/f398520f00a55ef12283d7563ee2624b49dbf2b6/eox_nelp/migrations/0001_initial.py#L83-L97
The reasons is that I want to test against like "`real certs data"`...

Reply:
So in the test, as you can see I don't mock any function of the certificates because the django test db has the model and data. So functions like GeneratedCertficates.objects... work and return data from test db.https://github.com/eduNEXT/eox-nelp/blob/f398520f00a55ef12283d7563ee2624b49dbf2b6/eox_nelp/stats/tests/tests_metrics.py#L273-L278

Reply:
Course overview was created because another model use that as foreign key, there was no another option

IMO there are another alternatives like Mock that could fit here without adding fields that you even don't use

Reply:
Actually, I started trying to use the mock approach, therefore as I used different Django queryset attributes and mixed them, I found that I have to do a lot of different mocks approaches and also with sense. So I stated to be a little mixed, so finally, I preferred to load the model and get data as the django queryset would work.
https://github.com/eduNEXT/eox-nelp/blob/f398520f00a55ef12283d7563ee2624b49dbf2b6/eox_nelp/stats/tests/tests_metrics.py#L257-L269

Also I set another fields, maybe extra but that could be useful if others created a certificate in the django db test model.

Actually all was loaded from here. Practically   I have the same model that the openedx
https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/certificates/models.py#L224-L240

Reply:
That's the point we can not copy all the models that openedx use